File: 2022/day06/day06.js
const input = require("fs").readFileSync(`${__dirname}/input.txt`).toString();

function findMarker(str, markerLength) {
  let index = 0;
  while (index++ < str.length) {
    let set = new Set();
    for (let ii = 0; ii < markerLength; ii++) {
      set.add(str[index + ii]);
    }
    if (markerLength == set.size) {
      return markerLength + index;
    }
  }
}

input.split("\n").map((line) => {
  console.log(`Part 1: ${findMarker(line, 4)}`);
  console.log(`Part 2: ${findMarker(line, 14)}`);
});

File: 2022/day06/day06.py
import os
input = open(os.path.dirname(__file__) + "/input.txt").read()

def find_marker(string, marker_length):
    index = 0
    while index < len(string):
      chars = set()
      for ii in range(marker_length):
        chars.add(string[index + ii])
      if marker_length == len(chars):
        return marker_length + index
      index += 1

for line in input.split("\n"):
  print("Part1: %d" % find_marker(line, 4))
  print("Part 2: %d" % find_marker(line, 14))
